pkg/controllermanager/controller/shoot/shoot_retry_control.go

```diff
@@ -52,63 +54,56 @@ func (c *Controller) shootRetryUpdate(oldObj, newObj interface{}) {
 	if shootFailedDueToRateLimits(newShoot) && !isShootFailed(oldShoot) {
 		key, err := cache.MetaNamespaceKeyFunc(newObj)
 		if err != nil {
-			gardenlogger.Logger.Errorf("Couldn't get key for object %+v: %v", newObj, err)
+			c.log.Error(err, "Couldn't get key for object", "object", newObj)
 			return
 		}
 		c.shootRetryQueue.Add(key)
 	}
 }
 
 // NewShootRetryReconciler creates a new instance of a reconciler which retries certain failed Shoots.
-func NewShootRetryReconciler(l logrus.FieldLogger, gardenClient client.Client, config *config.ShootRetryControllerConfiguration) reconcile.Reconciler {
+func NewShootRetryReconciler(gardenClient client.Client, config *config.ShootRetryControllerConfiguration) reconcile.Reconciler {
 	return &shootRetryReconciler{
-		logger:       l,
 		gardenClient: gardenClient,
 		config:       config,
 	}
 }
 
 type shootRetryReconciler struct {
-	logger       logrus.FieldLogger
 	gardenClient client.Client
 	config       *config.ShootRetryControllerConfiguration
 }
 
 func (r *shootRetryReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log := logf.FromContext(ctx)
+
 	shoot := &gardencorev1beta1.Shoot{}
 	if err := r.gardenClient.Get(ctx, request.NamespacedName, shoot); err != nil {
 		if apierrors.IsNotFound(err) {
-			r.logger.Infof("Object %q is gone, stop reconciling: %v", request.Name, err)
+			log.Info("Object is gone, stop reconciling")
 			return reconcile.Result{}, nil
 		}
-		r.logger.Infof("Unable to retrieve object %q from store: %v", request.Name, err)
-		return reconcile.Result{}, err
+		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
 	}
 
-	key := fmt.Sprintf("%s/%s", shoot.Namespace, shoot.Name)
-	shootLogger := r.logger.WithField("shoot", key)
-
 	if !shootFailedDueToRateLimits(shoot) {
 		return reconcile.Result{}, nil
 	}
 
 	mustRetry, requeueAfter := mustRetryNow(shoot, *r.config.RetryPeriod, r.config.RetryJitterPeriod)
 	if !mustRetry {
-		shootLogger.Infof("[SHOOT RETRY] Scheduled retry in %s", requeueAfter.Round(time.Second))
+		if requeueAfter == 0 {
+			return reconcile.Result{}, nil
+		}
+		log.V(1).Info("Scheduling retry for Shoot", "requeueAfter", requeueAfter.Round(time.Second))
 		return reconcile.Result{RequeueAfter: requeueAfter}, nil
 	}
 
-	shootLogger.Info("[SHOOT RETRY] Retrying a failed Shoot")
+	log.Info("Retrying failed Shoot")
 
-	shootCopy := shoot.DeepCopy()
+	patch := client.MergeFrom(shoot.DeepCopy())
 	metav1.SetMetaDataAnnotation(&shoot.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.ShootOperationRetry)
-	if err := r.gardenClient.Patch(ctx, shoot, client.MergeFrom(shootCopy)); err != nil {
-		return reconcile.Result{}, err
-	}
-
-	shootLogger.Info("[SHOOT RETRY] Shoot was successfully retried")
```

Comment:
Why is this log statement removed?

Reply:
Usually, we don't log about successful calls. The absence of errors will implicitly state that it was successfully retried.
Also, we already log what we are about to do earlier, so I found this log to be duplicated in a sense.